### ISSUE TITLE
[HELP] Are There any examples to verify?

### ISSUE THREAD

**Issue by oeular** [GoogleContainerTools/kpt-config-sync#5]
Here is my way  to install kpt-config-sync,
# Create a directory to store the manifest bundle
mkdir -p <MANIFEST_DIR>
# Extract the manifest bundle
tar -xzf <MANIFEST_BUNDLE> -C <MANIFEST_DIR>
# Apply the manifests to your cluster
kubectl apply -f <MANIFEST_DIR>

BUT Are There any efficient and easy example to verify the kpt-config-syn ?
I need some tips. Thank you !

**Comment by oeular** [comment]
@mortent  Hi, would you give me some suggestion ?

**Comment by oeular** [comment]
ping @justinsb Hello

**Comment by oeular** [comment]
@mortent  Thanks for your tips. Can the Config Sync work on the alone Kubernetes ? I want  to try it on the alone kubernetes.

**Comment by janetkuo** [comment]
@oeular yes Config Sync should work on Kubernetes clusters, including KinD.

**Comment by karlkfi** [comment]
Most of the examples we've invested in so far have been for Anthos Config Management in https://github.com/GoogleCloudPlatform/anthos-config-management-samples, but ACM is a higher level abstraction which includes Policy Controller, Config Sync, and an ACM Operator to manage them. Some of the examples in the samples repo work with CS by itself, but some require the rest of ACM. 

I would consider this a feature request for a set of Config Sync specific examples, perhaps in an /examples/ or /demos/ dir in this repo, like we've done for kpt and cli-utils.

**Comment by oeular** [comment]
@janetkuo  @karlkfi  Thanks. 
I need some specific examples on Kubernetes（We don't have ACM or Google Cloud）.
 Now I have installed Config Sync with https://github.com/GoogleContainerTools/kpt-config-sync/releases/download/launch/deployment-blueprint.tar.gz

Actually, I want to install Config Sync as an addon in Kubevela（https://github.com/kubevela/kubevela） for some component.

**Comment by tomasaschan** [comment]
(I'm piggy-backing on this existing issue instead of filing a new one; if you'd prefer I can file a separate one.)

Is there any docs or an example somewhere of how to use cluster selectors with the OSS manifests? After running `make build-oss` and copying the manifests (except for `dev.yaml` which I suppose should probably not have been there...?) into a folder of my own, it seems I can just apply all of them and deploy Config Sync. However, since the `ConfigManagement` is now out of the picture, I don't find anywhere how to specify the cluster name that then defines which labels etc that are applicable for cluster selectors in my target repositories. How does that work in the OSS version?

**Comment by karlkfi** [comment]
—cluster-name is configured on the reconciler-manager, which in turn configures it on the reconciler containers with an env var: https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/cmd/reconciler-manager/main.go#L42

**Comment by karlkfi** [comment]
FWIW, the `CLUSTER_NAME` environment variable can also be used, instead of the `--cluster-name` flag. 

It can be injected into the `reconciler-manager` container of the `reconciler-manager` Deployment or you can add an `envFrom` volume to populate env vars from a ConfigMap, which is what we do in our e2e tests.

---

As for non-GKE examples & docs, what exactly are you looking for @oeular ? 

Our public docs have improved considerably in the past few years, but they're mostly focused on Google customer use case, including GKE and non-GKE Fleet attached clusters, like AKS and EKS.
